### ISSUE TITLE
Unable to create circle name started from cyrillic letter

### ISSUE THREAD

**Issue by nbkgroup** [nextcloud/circles#207]
However, Cyrillic characters are not forbidden in names.

It makes the App compelely unusable for ordinary users.

**Comment by mathiasconradt** [comment]
Same for Arabic letter. Getting the error message:

Public circle 'تشكيل الحروف' could not be created: The name of your circle must start with an alpha-numerical character

![image](https://user-images.githubusercontent.com/295122/42202784-fabfde66-7e9c-11e8-90d6-c5c424754978.png)

**Comment by nbkgroup** [comment]
Looking at the code I see the cause of the error. The ad-hoc check for latin letters and numbers is hard coded in the CiclesController.php

$chars = 'abcdefghijklmnopqrstuvwxyz0123456789';
if (strpos($chars, strtolower(substr($name, 0, 1))) === false) {
	throw new CircleNameFirstCharException(

Unfortunately I now have no time to submerge to Nextcloud development nor properly installed environment to develop on PHP to fix that. I tried to replace it to language independent manner in my installation:

if (ctype_alnum(substr($name, 0, 1)) === false) {
	throw new CircleNameFirstCharException(

but it doesn't fix the problem. Look's like locale is not used for language support in NextCloud.

And, also, is there any reasonable reason to restrict first letter in that way?

**Comment by daita** [comment]
This was an old measure when the name of the circle was used to identify shares. I will see to remove this in a near future.

**Comment by mathiasconradt** [comment]
I removed the restriction in my test environment and I can create circles with arabic letters now, they are also found via auto-completion when trying to share.

Just one thing that is interesting: I have a self-signed SSL cert in my test environment. When I share to a group with Roman/Latin alphabet as name, no problem.

When I share to a group with Arabic letter in it, the share gets created, but I receive a cURL message about the self-signed cert. I tried multiple times and it definitely only happens when I share to a group with Arabic letters in it.

![image](https://user-images.githubusercontent.com/295122/42217655-59842546-7ec6-11e8-8a49-bb55faeffebc.png)
